src/Model/Database/DatabaseServices.cs

```
@@ -12,77 +12,91 @@ namespace Model.Database;
+        string surveyWrapperFile = Path.Combine(surveyWrapperPath, surveyWrapperId + ".json");
+        if (!File.Exists(surveyWrapperFile)) {
+            return null;
+        } else {
+            return LoadSurveyWrapperFromFile(surveyWrapperFile);
+        }
+    }
+
+    private static void SaveSurveyWrapperToFile(string surveyWrapperFilePath, SurveyWrapper surveyWrapper) {
+        string jsonString = JsonSerializer.Serialize(surveyWrapper);
+        File.WriteAllText(surveyWrapperFilePath, jsonString);
+    }
+
+    private string GetSurveyWrapperPath(int surveyWrapperId) {
+        return Path.Combine(databasePath, surveyWrapperId.ToString());
+    }
+
+    private static SurveyWrapper LoadSurveyWrapperFromFile(string surveyWrapperPath) {
+        string jsonString = File.ReadAllText(surveyWrapperPath);
+        return JsonSerializer.Deserialize<SurveyWrapper>(jsonString)!;
+    }
+
+    public string StorePictureOverwrite(int surveyWrapperId, string src) {
+        string surveyAssetsPath = GetSurveyWrapperAssetsPath(surveyWrapperId); 
         string dest = Path.Combine(surveyAssetsPath, Path.GetFileName(src));
         Directory.CreateDirectory(surveyAssetsPath);
         File.Copy(src, dest, true); //true -> overwrites automatically if dest already exists
+        return dest;
     }
 
-    public bool TryStorePicture(string src, int surveyId) {
-        string surveyAssetsPath = GetSurveyAssetsPath(surveyId);
+    public string TryStorePicture(int surveyWrapperId, string src) {
+        string surveyAssetsPath = GetSurveyWrapperAssetsPath(surveyWrapperId); 
         string dest = Path.Combine(surveyAssetsPath, Path.GetFileName(src));
         Directory.CreateDirectory(surveyAssetsPath);
         if (!File.Exists(dest)) {
             File.Copy(src, dest);
-            return true;
+            return dest;
         } else {
-            return false;
+            throw new Exception("A file with this name already exists for this surveyWrapper");
```

Do we want to throw an exception, or return an empty string? I think FrontEnd should decide.